adr/0003-PHPUnit-this-over-self.md

```diff
@@ -0,0 +1,31 @@
+# Use `$this` instead of `self` for PHPUnit assertions
+
+### Context
+
+PHPUnit assertions are static methods, yet in our code base we call them with `$this` instead of
+`self`.
+
+Whilst "incorrect", this usage does not break anything. Besides:
+
+- [PHUnit documentation][phpunit-doc] itself uses this by default
+- `$this` is much more widely used than `self` in this context in the community
+- all Infection code uses `$this`
+- `phpstan-strict-rules` itself (we use PHPStan in Infection) recommends this
```

Comment:
I doubt it's true. Are you sure?

Reply:
That comment was from yourself, I just took your word for it :P

Reply:
Hm... I don't think so :D

The only one place I remember is https://github.com/infection/infection/pull/1061#issuecomment-588964650 - and this is the opposite statement.

`phpstan-strict-rules` complains about `$this->assert***()`

I would suggest to remove this line